shared/js/contacts/import/utilities/overlay.js

```diff
@@ -1,40 +1,151 @@
-      progressMsg = document.querySelector('#progress-msg'),
-      menu = document.querySelector('#loading-overlay menu'),
-      cancelButton = document.querySelector('#cancel-overlay');
-
-  // Constructor for the progress element
-  function ProgressBar(pMsgId, pClass) {
-    var counter = 0;
-    var total = 0;
-    var progressTextId = pMsgId || 'genericProgress';
-    var clazz = pClass;
+/* global LazyLoader, HtmlImports */
 
+(function(exports) {
+  'use strict';
+
+  var counter = 0;
+  var total = 0;
+  var progressTextId;
+  var progressType;
+
+  var overlay,
+      statusContainer,
+      progressActivity,
+      progressTitle,
+      progressElement,
+      progressMsg,
+      menu,
+      cancelButton,
+      _cancelCB,
+      isAnimationPlaying = false,
+      isShown = false;
+
+  var overlayReady,
+      _reject,
+      isLoading = false;
+
+  var link = document.createElement('link');
+  link.setAttribute('rel', 'import');
+  link.setAttribute('href', '/shared/elements/contacts/overlay.html');
+
+  function cacheElements() {
+    statusContainer = overlay.querySelector('p[role="status"]'),
+    progressActivity = document.querySelector('#progress-activity'),
+    progressTitle = document.querySelector('#progress-title'),
+    progressElement = document.querySelector('#progress-element'),
+    progressMsg = document.querySelector('#progress-msg'),
+    menu = document.querySelector('#loading-overlay menu'),
+    cancelButton = document.querySelector('#cancel-overlay');
     progressElement.setAttribute('value', 0);
+  }
 
-    function showMessage() {
+  function addListeners() {
+    cancelButton.onclick = function on_cancel(evt) {
+      evt.preventDefault();
+      delete cancelButton.onclick;
+      cancelButton.disabled = false;
+      if (typeof _cancelCB === 'function') {
+        _cancelCB();
+      }
+      _cancelCB = null;
+    };
+  }
+
+  function deleteOverlay() {
+    document.head.removeChild(link);
+    document.body.removeChild(overlay);
+    overlay = null;
+    overlayReady = null;
+    _reject = null;
+    isLoading = false;
+    counter = 0;
+    total = 0;
+    if (cancelButton) {
+      delete cancelButton.onclick;
+    }
+  }
+
+  function populatePanel() {
+    overlay = document.createElement('form');
+    overlay.setAttribute('is', 'confirm-form');
+    overlay.setAttribute('data-type', 'confirm');
+    overlay.setAttribute('id', 'loading-overlay');
+    overlay.setAttribute('role', 'dialog');
+    overlay.className = 'hide no-opacity';
+    document.head.appendChild(link);
+    document.body.appendChild(overlay);
+  }
+
+  function init() {
+    // To prevent operation over the elements when they're still being loaded
+    overlayReady = new Promise(function(resolve, reject) {
+      isLoading = true;
+      _reject = reject;
+      populatePanel();
+      LazyLoader.load([
+        '/shared/js/html_imports.js'
+      ], function() {
```

Comment:
`LazyLoader.load` can return a Promise as well, Why dont we return this promise instead? Could you try to use the LazyLoader for loading the panel, instead of using populate?